testing/todocheck_test.go

```diff
@@ -742,6 +742,55 @@ func TestPrintingVersionFlagStopsProgram(t *testing.T) {
 	}
 }
 
+func TestVueTodos(t *testing.T) {
+	err := scenariobuilder.NewScenario().
+		WithBinary("../todocheck").
+		WithBasepath("./scenarios/vue").
+		WithConfig("./test_configs/no_issue_tracker.yaml").
+		WithIssueTracker(issuetracker.Jira).
+		WithIssue("1", issuetracker.StatusOpen).
+		WithIssue("2", issuetracker.StatusClosed).
+		WithIssue("3", issuetracker.StatusOpen).
+		WithIssue("4", issuetracker.StatusOpen).
+		ExpectTodoErr(
+			scenariobuilder.NewTodoErr().
+				WithType(errors.TODOErrTypeMalformed).
+				WithLocation("scenarios/vue/main.vue", 1).
+				ExpectLine("// oneline comment, malformed TODO")).
+		ExpectTodoErr(
+			scenariobuilder.NewTodoErr().
+				WithType(errors.TODOErrTypeIssueClosed).
+				WithLocation("scenarios/vue/main.vue", 3).
+				ExpectLine("// TODO 2: oneline comment with Issue closed")).
+		ExpectTodoErr(
+			scenariobuilder.NewTodoErr().
+				WithType(errors.TODOErrTypeIssueClosed).
+				WithLocation("scenarios/vue/main.vue", 6)).
+		ExpectTodoErr(
+			scenariobuilder.NewTodoErr().
+				WithType(errors.TODOErrTypeMalformed).
+				WithLocation("scenarios/vue/main.vue", 7)).
+		ExpectTodoErr(
+			scenariobuilder.NewTodoErr().
+				WithType(errors.TODOErrTypeIssueClosed).
+				WithLocation("scenarios/vue/main.vue", 10)).
+		ExpectTodoErr(
+			scenariobuilder.NewTodoErr().
+				WithType(errors.TODOErrTypeMalformed).
+				WithLocation("scenarios/vue/main.vue", 0).
+				ExpectLine("TODO: malformed CS/JS multline entry, missing number").
```

Comment:
Why is the location of this line `0`?

Reply:
Shouldn't this be:
```go
WithLocation("scenarios/vue/main.vue", 15).
ExpectLine("/*").
ExpectLine("TODO: ...").
ExpectLine("*/").
```

Reply:
Oh, this is weird. I mean, you are right, but I get an error if I implement it like you proposed:

`/todocheck/testing/todocheck_test.go:791: No matching todo detected in any of the scenarios
        
        Actual:
        ERROR: Malformed todo
        scenarios/vue/main.vue:0: TODO: malformed CS/JS multline entry, missing number
        scenarios/vue/main.vue:1: */
        	> TODO should match pattern - TODO {task_id}:
        
        Remaining scenarios:
        (scenario #1)
        ERROR: Malformed todo
        scenarios/vue/main.vue:15: /*
        scenarios/vue/main.vue:16: TODO: ...
        scenarios/vue/main.vue:17: */
        	> TODO should match pattern - TODO {task_id}:
        
        (scenario #2)
        ERROR: Issue is closed
        scenarios/vue/main.vue:49:   color: blue; // TODO 2: online comment wellformed in code, BUT issue closed
        `
        
And here, the displayed line of the actual todo is zero. I'm confused. Do you have any idea why it behaves that way?

Reply:
This actually is a bug with the comment matcher implementation.

There are two issues:
 * You are not matching the multiline css comments correctly. You should match `/* some todo comment */`, but instead you are matching `some todo comment */`
 * You are not tracking the linecnt at which a multiline comment begins. It should be persisted when transitioning to the multiline comment state similar to [how it's done here](https://github.com/preslavmihaylov/todocheck/blob/master/matchers/standard/comments.go#L37) and [later used here](https://github.com/preslavmihaylov/todocheck/blob/master/matchers/standard/comments.go#L89)

Reply:
You can try reproducing this by:
 * `go build todocheck` and copying the output binary to a comfortable location
 * Adding a new file `main.vue` which contains the scenario from the test
 * Add a new `.todocheck.yaml` file with contents:
```
origin: https://github.com/todocheck-tests/test-public
issue_tracker: GITHUB
```
 * run `./todocheck` & verify the results

You can try instrumenting some debug prints to see what's going on or use a standard debugger if you have one set up.

Reply:
Hey, you were right, it was indeed a bug. I think the new commits I pushed fix it. At least, now all the testing works as expected. I also added another test for an invalid HTML multiline TODO, which also works. 

But there is one other thing I noticed, and I am not sure if its a feature or a bug: The last line of every file is always ignored, also by other file matchers (checked PHP briefly). This means that multiline comments that are closed in the last line are not checked for errors (since they technically never become closed). 

If this is indeed a feature, I'd be interested in an explanation for it. Maybe it would then be helpful to throw an error if the last line is not empty, since it could lead to missing TODOs otherwise?

Reply:
> But there is one other thing I noticed, and I am not sure if its a feature or a bug: The last line of every file is always 
> ignored, also by other file matchers (checked PHP briefly). This means that multiline comments that are closed in the
> last line are not checked for errors (since they technically never become closed).

The final testcase you've added:
```go
		ExpectTodoErr(
			scenariobuilder.NewTodoErr().
				WithType(errors.TODOErrTypeMalformed).
				WithLocation("scenarios/vue/main.vue", 53).
				ExpectLine("<!--").
				ExpectLine("TODO : malformed HTML multiline entry").
				ExpectLine("-->")).
```

proves it otherwise. What exactly do you mean?

If your observation is indeed correct, then this is a bug. If you can reproduce it deterministically, could you open a bug with reproduction steps & share the details?